pkgs/stdenv/generic/check-meta.nix

```diff
@@ -175,8 +175,9 @@ let
   checkMeta = meta: if shouldCheckMeta then lib.remove null (lib.mapAttrsToList checkMetaAttr meta) else [];
 
   checkPlatform = attrs:
-    (!(attrs ? meta.platforms) || lib.any (lib.meta.platformMatch hostPlatform) attrs.meta.platforms) &&
-    (!(attrs ? meta.badPlatforms && lib.any (lib.meta.platformMatch hostPlatform) attrs.meta.badPlatforms));
+    (lib.any (lib.meta.platformMatch hostPlatform)
+             (attrs.meta.platforms or lib.platforms.all)) &&
+    (! lib.elem hostPlatform.system (attrs.meta.badPlatforms or []));
```

Comment:
This is definitely wrong, we want platformMatch semantics here too right?

Reply:
Oh yes, I didn't see the `rec` and therefore misinterpreted `badPlatforms = platforms;` .